---
title: Australian Stingless Bee Nest Microbiome
subtitle: |
  Supplementary material for the paper *16S Amplicon Metabarcoding from Nest 
  Materials of Native Australian Stingless Bees*
date: 2022-11-10
---

[![License: CC BY 4.0](https://img.shields.io/badge/License-CC_BY_4.0-lightgrey.svg)](LICENSE.md)
[![DOI](https://zenodo.org/badge/517895941.svg)](https://zenodo.org/badge/latestdoi/517895941)

## Introduction

This repository contains the code and documentation required to reproduce the 
results presented in our paper "16S Amplicon Metabarcoding from Nest Materials 
of Native Australian Stingless Bees". We describe how the raw metabarcoding 
datasets were processed using QIIME 2 and further analysed in R to produce the 
figure included in our paper. Beta diversity statistics describing microbiome 
variability associated with the different bee species and material types sampled 
are calculated, visualised, and used for hypothesis testing.

## Data Availability

All data are available from NCBI SRA with study accession 
[SRP405832](https://trace.ncbi.nlm.nih.gov/Traces/?view=study&acc=SRP405832).

Samples were sequenced in two batches, identified by the last two characters of 
each sample name.

## Processing in `QIIME 2`

Raw sequence data were processed on the QUT HPC Lyra. Batches B1 and B2 were 
analysed separately using the following scripts: 
[B1](Data/HPC_Scripts/MRA_QIIME_Script.sh), 
[B2](Data/HPC_Scripts/MRA_QIIME_Script.sh).

## Setup

Load packages and pre-defined functions.

```{r setup, warning=FALSE, message=FALSE}
library(phyloseq)
library(qiime2R)
library(tidyverse)
library(magrittr)
library(symbioteR)
library(vegan)
library(ggforce)
library(hrbrthemes)
library(concaveman)
library(ggh4x)
library(ggtext)
library(svglite)
source("functions.R")
```

## Load Data

We read the .QZA artifacts as phyloseq objects, merge them, and add metadata.

```{r, message=FALSE}
hiveMB_1 <- readQZAFolder("Data/H1")
hiveMB_2 <- readQZAFolder("Data/H2")

hiveMB <- merge_phyloseq(hiveMB_1, hiveMB_2)

sample_data(hiveMB) <- read_tsv("Data/Metadata.tsv") |>
  mutate(Year = str_extract(Date_Harvest, "\\d{4}$")) |> 
  column_to_rownames("ID")
```

## Filter Samples

We remove any mitochondrial or chloroplast ASVs, and any that were unclassified 
at the kingdom level.

```{r}
hiveMB <- removeNonbacterial(hiveMB)
```

## Beta Diversity

### Perform Ordination

Convert to relative abundance and perform NDMS ordination of Bray-Curtis 
dissimilarities.

```{r, results='asis'}
set.seed(42)

hiveMBRel <- hiveMB |> 
  makeObservationsRelative()

ord <- ordinate(hiveMBRel, 
                distance = "bray", 
                method = "NMDS", 
                k = 4, 
                maxit = 1000,
                trace = 0)

ord
```

### Plot Ordination

Plot the NMDS ordination.

```{r, warning=FALSE}
custPal <- viridis::turbo(n = 4)[c(1:2, 4)]

ordPlot <- plot_ordination(hiveMBRel, ord, justDF = TRUE) |> 
    ggplot(data = _, 
           aes(x = NMDS2,
               y = NMDS1,
               shape = Material, 
               color = Bees)) +
    geom_point(size = 3, alpha = 0.8) +
    theme_ipsum(base_family = "sans") +
    geom_mark_hull(aes(x = NMDS2, 
                       y = NMDS1,
                       x0 = NMDS2,
                       y0 = NMDS1,
                       fill = Year, 
                       label = Year),
                   inherit.aes = FALSE,
                   label.margin = margin(),
                   con.cap = 0,
                   label.fill = NA,
                   concavity = 3,
                   alpha = 0.2) +
    scale_colour_manual(values = custPal, "Species",
                        labels = c(expression(paste(italic("A. australis"))),
                                   expression(paste(italic("T. carbonaria"))),
                                   expression(paste(italic("T. hockingsi"))))) +
    scale_fill_brewer(palette = "Pastel2", guide = "none") +
    scale_shape_manual(values = c(15, 16, 17, 18)) +
    theme(legend.text.align = 0,
          legend.position = "bottom",
          legend.box = "vertical",
          legend.margin = margin(),
          text = element_text(family = "sans"),
          aspect.ratio = 3.5/4.5) +
    scale_x_continuous(expand = expansion(add = 0.5)) +
    scale_y_continuous(expand = expansion(add = 0.5))

ggsave("Out/Figures/Figure1.svg", ordPlot, device = "svg", bg = "white", 
       width = 6.5, height = 6, dpi = 1200)
```

!["Figure 1: NMDS Ordination"](Out/Figures/Figure1.png)

### Significance Test

Perform a PERMANOVA test.

```{r}
#| tbl-cap: "Table 1: PERMANOVA Results"

dist <- phyloseq::distance(hiveMB, method = "bray")
adonis2(dist ~ Bees + Material + Year + Hive_ID, 
                as(sample_data(hiveMB), "data.frame"))
```

## Taxonomic Abundance

### Key Genera

We highlight the abundance of a number of key taxonomic groups referenced in 
the paper.

```{r, message=FALSE}
abPlot <- hiveMBRel |> 
  tax_glom("Genus") |> 
  psmelt() |> 
  group_by(Genus, Material, Bees) |> 
  mutate(Bees = case_when(Bees == "Austroplebia australis" ~ "A. australis",
                          Bees == "Tetragonula carbonaria" ~ "T. carbonaria",
                          Bees == "Tetragonula hockingsi" ~ "T. hockingsi")) |> 
  summarise(Proportion = mean(Abundance)) |> 
  filter(Genus %in% c("Lactobacillus", "Bombella", "Gilliamella",
                      "Snodgrassella")) |> 
  ggplot(aes(x = Material, y = Genus, fill = Proportion, 
              label = round(Proportion, 2))) +
    geom_tile() +
    geom_text(colour = "white") +
    facet_wrap(~ Bees) +
    theme_ipsum(base_family = "sans") +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
          strip.text = element_text(face = "italic"),
          legend.position = "none")

ggsave("Out/Figures/Figure2.png", abPlot, device = "png", bg = "white", 
       width = 8, height = 4, dpi = 1200)
```

!["Figure 2: Average Proportion"](Out/Figures/Figure2.png)

### Gammaproteobacteria

We note the abundance of ASVs from the class Gammaproteobacteria.

```{r, message=FALSE}
#| tbl-cap: "Table 2: Mean Proportion of Gammaproteobacteria"

hiveMBRel |> 
  tax_glom("Class") |> 
  psmelt() |> 
  group_by(Bees, Material, Class) |>  
  summarise(Proportion = round(mean(Abundance), 2)) |> 
  filter(Class == "Gammaproteobacteria") |> 
  select(!Class)
```

## Export Data

We provide a copy of the `phyloseq` object that can be easily imported for 
further analysis in R.

```{r}
saveRDS(hiveMB, "Out/hiveMB.rds")
```

[Download it here.](Out/hiveMB.rds)

## Session Info

```{r}
sessionInfo()
```
